File: internal/models/location.go
package models

import (
	"github.com/google/uuid"
	"time"
)

type Location struct {
	ID           uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primary_key"`
	Name         string    `gorm:"type:varchar(255);not null"`
	Slug         string    `gorm:"type:varchar(20);not null;uniqueIndex"`
	Address      string    `gorm:"type:varchar(255);not null"`
	PinballMapID int       `gorm:"type:int;not null"`
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

File: web/app/src/test/fake/location.ts
import {Location, pinballMap} from "../../api";
import {faker} from "@faker-js/faker";
import * as helpers from "../../helpers";

export function location(): Location {
  const name = faker.hacker.phrase();

  return {
    id: faker.datatype.uuid(),
    name: name,
    slug: helpers.slugify(name),
    address: faker.address.streetAddress(),
    pinball_map_id: faker.datatype.number(),
    created_at: faker.date.recent(5).toISOString(),
    updated_at: faker.date.recent(1).toISOString()
  }
}

export function pinballMapLocation(): pinballMap.Location {
  return {
    id: faker.datatype.number(),
    name: faker.hacker.phrase(),
    street: faker.address.streetAddress(),
    city: faker.address.city(),
    state: faker.address.stateAbbr(),
    zip: faker.address.zipCode(),
    country: faker.address.countryCode(),
    num_machines: faker.datatype.number(),
  }
}
